src/Umbraco.Core/Migrations/Upgrade/V_8_0_0/DataTypeMigration.cs

```diff
@@ -60,43 +61,42 @@ public override void Migrate()
-                if (aliases.Length == 1 && string.IsNullOrWhiteSpace(aliases[0]))
+                // migrate the preValues to configuration
+                var migrator = _preValueMigrators.GetMigrator(dataType.EditorAlias) ?? new DefaultPreValueMigrator();
+                var config = migrator.GetConfiguration(dataType.NodeId, dataType.EditorAlias, group.ToDictionary(x => x.Alias, x => x));
+                var json = JsonConvert.SerializeObject(config);
+
+                // validate - and kill the migration if it fails
+                var newAlias = migrator.GetNewAlias(dataType.EditorAlias);
+                if (newAlias == null)
+                {
+                    _logger.Warn<DataTypeMigration>($"Skipping validation of configuration for data type {dataType.NodeId} : {dataType.EditorAlias}."
+                                                    + $" Please ensure that the configuration is valid. The site may fail to start and / or load data types and run.");
+                }
+                else if (!_propertyEditors.TryGet(newAlias, out var propertyEditor))
                 {
-                    // array-based prevalues
-                    var values = new Dictionary<string, object> { ["values"] = group.OrderBy(x => x.SortOrder).Select(x => x.Value).ToArray() };
-                    dataType.Configuration = JsonConvert.SerializeObject(values);
+                    _logger.Warn<DataTypeMigration>($"Skipping validation of configuration for data type {dataType.NodeId} : {newAlias}{(dataType.EditorAlias == newAlias ? "" : $" ({dataType.EditorAlias})")}"
+                                                    + $" because no property editor with alias {newAlias} was found."
+                                                    + $" Please ensure that the configuration is valid. The site may fail to start and / or load data types and run.");
                 }
                 else
                 {
-                    // assuming we don't want to fall back to array
-                    if (aliases.Length != group.Count() || aliases.Any(string.IsNullOrWhiteSpace))
-                        throw new InvalidOperationException($"Cannot migrate datatype w/ id={dataType.NodeId} preValues: duplicate or null/empty alias.");
-
-                    // dictionary-base prevalues
-                    var values = group.ToDictionary(x => x.Alias, x => x.Value);
-                    dataType.Configuration = JsonConvert.SerializeObject(values);
+                    var configEditor = propertyEditor.GetConfigurationEditor();
+                    try
+                    {
+                        var _ = configEditor.FromDatabase(json);
+                    }
+                    catch (Exception e)
+                    {
+                        _logger.Warn<DataTypeMigration>($"Failed to validate configuration for data type {dataType.NodeId} : {newAlias}{(dataType.EditorAlias == newAlias ? "" : $" ({dataType.EditorAlias})")}."
```

Comment:
Any reason to not use structured logging?

Reply:
Also please don't introduce the unused variable e, either use  catch (Exception) or use e in the log.

Reply:
fixing